testing/file_test/autoupdate.h

```diff
@@ -197,17 +209,23 @@ class FileTestAutoupdater {
   std::string test_command_;
   std::string dump_command_;
   llvm::StringRef input_content_;
-  const llvm::SmallVector<llvm::StringRef>& filenames_;
   int autoupdate_line_number_;
   bool autoupdate_split_;
   const llvm::SmallVector<FileTestLine>& non_check_lines_;
   const std::optional<RE2>& default_file_re_;
   const llvm::SmallVector<LineNumberReplacement>& line_number_replacements_;
   std::function<auto(std::string&)->void> do_extra_check_replacements_;
 
+  // If `autoupdate_split_` is true, the file number of the autoupdate split.
+  // Otherwise, meaningless.
```

Comment:
WDYT of setting `-1` when there is no split, and removing `autoupdate_split_`? I'm worried about confusion here, given they're relating similar information.

(you could also choose to replace `autoupdate_split` in the constructor)

Reply:
Done -- changed to `optional<int>` and used `nullopt` instead of -1.